datalad/cmd.py

```diff
@@ -98,7 +98,208 @@ def _cleanup_output(stream, std):
+          Environment to be passed to subprocess.Popen(). If `cwd`
+          was given, 'PWD' in the environment is set to its value.
+          This must be a complete environment definition, no values
+          from the current environment will be inherited.
+        poll_period : float, optional
+          Interval at which the running process is queried for
+          output.
+        """
+        self._poll_period = poll_period
+        self.env = env.copy() if env else None
+        # stringify to support Path instances on PY35
+        self.cwd = str(cwd) if cwd is not None else None
+        if self.cwd and env is not None:
+            # if CWD was provided, we must not make it conflict with
+            # a potential PWD setting
+            self.env['PWD'] = self.cwd
+
+    def run(self, cmd, proc_stdout=None, proc_stderr=None, stdin=None):
+        """Execute a command and communicate with it.
+
+        Parameters
+        ----------
+        cmd : list
+          Sequence of program arguments. Passing a single string means
+          that it is simply the name of the program, no complex shell
+          commands are supported.
+        proc_stdout : callable, optional
+          By default no stdout is captured, but relayed to the parent
+          process's stdout. If given, all stdout is passed as a byte-string
+          to this callable, in the chunks it was received by polling the
+          processing. The callable may transform it in any way, its output
+          (byte-string) is concatenated and provided as stdout return value.
+          The helper functions 'kill_output' and 'capture_output' are
+          provided to either swallow all output (and not relay it to the
+          parent) or to capture all output and provide it as the return
+          value.
+        proc_stderr : callable, optional
+          Like proc_stdout, but for stderr.
+        stdin : byte stream, optional
+          File descriptor like, used as stdin for the process. Passed
+          verbatim to subprocess.Popen().
+
+        Returns
+        -------
+        stdout, stderr
+          Unicode string with the cumulative standard output and error
+          of the process.
+
+        Raises
+        ------
+        CommandError
+          On execution failure (non-zero exit code) this exception is
+          raised which provides the command (cmd), stdout, stderr,
+          exit code (status), and a message identifying the failed
+          command, as properties.
+        """
+        proc_out = (proc_stdout, proc_stderr)
+        if all(p is None for p in proc_out):
+            proc_out = None
+        try:
+            lgr.log(8, "Start running %r", cmd)
+            process = subprocess.Popen(
+                cmd,
+                # from PY37 onwards
+                #capture_output=proc_output,
+                stdout=subprocess.PIPE if proc_out else None,
+                stderr=subprocess.PIPE if proc_out else None,
```

Comment:
I'm confused about why, if one stream is being captured, that means the other stream should (i.e. about why it's `if proc_out` instead of `if proc_{stdout,stderr}`).  Assuming there's a good reason for this, the interaction should be documented for the proc_stdout/proc_stderr parameters.

Reply:
I cannot come up with a reason (anymore), will change.

Reply:
Done